block/pruning.go

```diff
@@ -3,35 +3,35 @@ package block
 import (
 	"context"
 	"fmt"
-
-	"github.com/dymensionxyz/gerr-cosmos/gerrc"
 )
 
-func (m *Manager) PruneBlocks(retainHeight uint64) error {
+func (m *Manager) PruneBlocks(retainHeight uint64) (uint64, error) {
 	if m.IsProposer() && m.NextHeightToSubmit() < retainHeight { // do not delete anything that we might submit in future
-		return fmt.Errorf("cannot prune blocks before they have been submitted: retain height %d: next height to submit: %d: %w",
-			retainHeight,
-			m.NextHeightToSubmit(),
-			gerrc.ErrInvalidArgument)
+		m.logger.Debug("cannot prune blocks before they have been submitted. using height last submitted height for pruning", "retain_height", retainHeight, "height_to_submit", m.NextHeightToSubmit())
+		retainHeight = m.NextHeightToSubmit() - 1
```

Comment:
I think there is a race here because you read m.NextHeightToSubmit() twice it can change in between

Reply:
also isnt this off by one? why isn't retain height just m.NextHeightToSubmit(?

Reply:
fixed